MSAL/test/app/mac/MSALAcquireTokenViewController.m

```diff
@@ -0,0 +1,366 @@
+        NSAlert *alert = [[NSAlert alloc] init];
+        alert.messageText = messageText;
+        [alert addButtonWithTitle:@"OK"];
+        alert.informativeText = informativeText;
+        [alert runModal];
+    });
+}
+
+- (IBAction)clearCache:(id)sender
+{
+    MSALTestAppSettings *settings = [MSALTestAppSettings settings];
+    
+    // Delete accounts.
+    NSDictionary *currentProfile = [MSALTestAppSettings currentProfile];
+    NSString *clientId = [currentProfile objectForKey:MSAL_APP_CLIENT_ID];
+    NSString *redirectUri = [currentProfile objectForKey:MSAL_APP_REDIRECT_URI];
+    __auto_type authority = [settings authority];
+    
+    NSError *error = nil;
+    MSALPublicClientApplication *application = [[MSALPublicClientApplication alloc] initWithClientId:clientId
+                                                                                           authority:authority
+                                                                                         redirectUri:redirectUri
+                                                                                               error:&error];
+    
+    BOOL result = [application.tokenCache clearWithContext:nil error:&error];
+    
+    if (result)
+    {
+        [self.resultView setString:@"Successfully cleared cache."];
+        settings.currentAccount = nil;
+        
+        [[NSNotificationCenter defaultCenter] postNotificationName:MSALTestAppCacheChangeNotification object:self];
+    }
+    else
+    {
+        [self.resultView setString:[NSString stringWithFormat:@"Failed to clear cache, error = %@", error]];
+    }
+}
+
+- (IBAction)clearCookies:(id)sender
+{
+    NSHTTPCookieStorage* cookieStore = [NSHTTPCookieStorage sharedHTTPCookieStorage];
+    NSArray* cookies = cookieStore.cookies;
+    for (NSHTTPCookie* cookie in cookies)
+    {
+        [cookieStore deleteCookie:cookie];
+    }
+    
+    [_resultView setString:[NSString stringWithFormat:@"Cleared %lu cookies.", (unsigned long)cookies.count]];
+    
+    // Clear WKWebView cookies
+    if (@available(macOS 10.11, *)) {
+        NSSet *allTypes = [WKWebsiteDataStore allWebsiteDataTypes];
+        [[WKWebsiteDataStore defaultDataStore] removeDataOfTypes:allTypes
+                                                   modifiedSince:[NSDate dateWithTimeIntervalSince1970:0]
+                                               completionHandler:^{}];
+    }
+}
+
+- (IBAction)acquireTokenInteractive:(id)sender
+{
+    (void)sender;
+    
+    MSALTestAppSettings *settings = [MSALTestAppSettings settings];
+    NSDictionary *currentProfile = [MSALTestAppSettings currentProfile];
+    NSString *clientId = [currentProfile objectForKey:MSAL_APP_CLIENT_ID];
+    NSString *redirectUri = [currentProfile objectForKey:MSAL_APP_REDIRECT_URI];
+    MSALAuthority *authority = [settings authority];
+    NSDictionary *extraQueryParameters = [NSDictionary msidDictionaryFromWWWFormURLEncodedString:[self.extraQueryParamsField stringValue]];
```

Comment:
does it mean we need to enter EQP as url encoded string? seems pretty complicated...

Reply:
Still having the same question. What is the user expected to enter to the EQP field?